### ISSUE TITLE
Unable to change between hugepages

### ISSUE THREAD

**Issue by SchSeba** [openshift-kni/performance-addon-operators#181]
Reporduce:
deploy performance CR
```
spec:
  cpu:
    balanceIsolated: false
    isolated: 0-15
    reserved: 16-40
  hugepages:
    defaultHugepagesSize: 2M
    pages:
    - count: 16
      node: 0
      size: 2M
  nodeSelector:
    node-role.kubernetes.io/worker-cnf: ""
  realTimeKernel:
    enabled: false
```

Then wait for the policy to be applied and update the policy again this time change the hugepages

```
spec:
  cpu:
    balanceIsolated: false
    isolated: 0-15
    reserved: 16-40
  hugepages:
    defaultHugepagesSize: 1G
    pages:
    - count: 16
      node: 0
      size: 1G
  nodeSelector:
    node-role.kubernetes.io/worker-cnf: ""
  realTimeKernel:
    enabled: false
```

The node will reboot but never become ready.
I check the kubelet logs and find this.

```
Apr 19 16:16:33 cnfd0-worker-0.fci1.kni.lab.eng.bos.redhat.com hyperkube[122892]: E0419 16:16:33.986881  122892 kubelet_node_status.go:402] Error updating node status, will retry: failed to patch status "{\"status\":{\"$setElementOrder/conditions\":[{\"type\":\"MemoryPressure\"},{\"type\":\"DiskPressure\"},{\"type\":\"PIDPressure\"},{\"type\":\"Ready\"}],\"allocatable\":{\"cpu\":\"65\",\"hugepages-1Gi\":\"16Gi\",\"memory\":\"376820060Ki\"},\"capacity\":{\"hugepages-1Gi\":\"16Gi\"},\"conditions\":[{\"lastHeartbeatTime\":\"2020-04-19T16:16:33Z\",\"lastTransitionTime\":\"2020-04-19T16:16:33Z\",\"message\":\"kubelet has sufficient memory available\",\"reason\":\"KubeletHasSufficientMemory\",\"status\":\"False\",\"type\":\"MemoryPressure\"},{\"lastHeartbeatTime\":\"2020-04-19T16:16:33Z\",\"lastTransitionTime\":\"2020-04-19T16:16:33Z\",\"message\":\"kubelet has no disk pressure\",\"reason\":\"KubeletHasNoDiskPressure\",\"status\":\"False\",\"type\":\"DiskPressure\"},{\"lastHeartbeatTime\":\"2020-04-19T16:16:33Z\",\"lastTransitionTime\":\"2020-04-19T16:16:33Z\",\"message\":\"kubelet has sufficient PID available\",\"reason\":\"KubeletHasSufficientPID\",\"status\":\"False\",\"type\":\"PIDPressure\"},{\"lastHeartbeatTime\":\"2020-04-19T16:16:33Z\",\"lastTransitionTime\":\"2020-04-19T16:16:33Z\",\"message\":\"kubelet is posting ready status\",\"reason\":\"KubeletReady\",\"status\":\"True\",\"type\":\"Ready\"}],\"nodeInfo\":{\"bootID\":\"b35a3754-8f9f-4111-84c8-546cedb79a72\"}}}" for node "cnfd0-worker-0.fci1.kni.lab.eng.bos.redhat.com": Node "cnfd0-worker-0.fci1.kni.lab.eng.bos.redhat.com" is invalid: [status.capacity.hugepages-2Mi: Invalid value: resource.Quantity{i:resource.int64Amount{value:10485760, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"10Mi", Format:"BinarySI"}: may not have pre-allocated hugepages for multiple page sizes, status.capacity.openshift.io/sriovnic: Invalid value: resource.Quantity{i:resource.int64Amount{value:0, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"0", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.capacity.openshift.io/testresource: Invalid value: resource.Quantity{i:resource.int64Amount{value:5, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"5", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.capacity.pods: Invalid value: resource.Quantity{i:resource.int64Amount{value:250, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"250", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.capacity.memory: Invalid value: resource.Quantity{i:resource.int64Amount{value:404207529984, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"394733916Ki", Format:"BinarySI"}: may not have pre-allocated hugepages for multiple page sizes, status.capacity.ephemeral-storage: Invalid value: resource.Quantity{i:resource.int64Amount{value:478926602240, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"456740Mi", Format:"BinarySI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.hugepages-1Gi: Invalid value: resource.Quantity{i:resource.int64Amount{value:17179869184, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"", Format:"BinarySI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.pods: Invalid value: resource.Quantity{i:resource.int64Amount{value:250, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"250", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.cpu: Invalid value: resource.Quantity{i:resource.int64Amount{value:65, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"65", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.ephemeral-storage: Invalid value: resource.Quantity{i:resource.int64Amount{value:429960199479, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"429960199479", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.openshift.io/apivolresource: Invalid value: resource.Quantity{i:resource.int64Amount{value:0, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"0", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.openshift.io/sriovnic: Invalid value: resource.Quantity{i:resource.int64Amount{value:0, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"0", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.openshift.io/testresource: Invalid value: resource.Quantity{i:resource.int64Amount{value:5, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"5", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.openshift.io/mturesource: Invalid value: resource.Quantity{i:resource.int64Amount{value:0, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"0", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.memory: Invalid value: resource.Quantity{i:resource.int64Amount{value:385863741440, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"376820060Ki", Format:"BinarySI"}: may not have pre-allocated hugepages for multiple page sizes, status.allocatable.openshift.io/dpdknic: Invalid value: resource.Quantity{i:resource.int64Amount{value:5, scale:0}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"5", Format:"DecimalSI"}: may not have pre-allocated hugepages for multiple page sizes]
Apr 19 16:16:33 cnfd0-worker-0.fci1.kni.lab.eng.bos.redhat.com hyperkube[122892]: E0419 16:16:33.986896  122892 kubelet_node_status.go:389] Unable to update node status: update node status exceeds retry count

```

**Comment by slintes** [comment]
This is a known limitation atm: you need to remove the hp config first, wait, and then apply the new config, until some upstream k8s issues are fixed.
Keeping this open as we should document this at least, and maybe catch this in the validation code.
/cc @cynepco3hahue fyi

**Comment by cynepco3hahue** [comment]
The u/s issue fixed and all should work as expected under 4.6.
